realm/realm-library/src/androidTest/kotlin/io/realm/DictionaryTests.kt

```diff
@@ -300,52 +317,118 @@ class DictionaryTests {
+                }
+            }
 
-            val instance = realm.where<DictionaryClass>()
-                    .findFirst()
-            assertNotNull(instance)
+            // TODO: parameterize to test all types
 
-            with(instance) {
-                // All types but byte[] can be asserted like this
-                myBooleanDictionary!!.putAndAssert(KEY_HELLO, VALUE_HELLO)
-                myStringDictionary!!.putAndAssert(KEY_HELLO, VALUE_HELLO_STRING)
-                myIntegerDictionary!!.putAndAssert(KEY_HELLO, VALUE_HELLO_NUMERIC)
-                myFloatDictionary!!.putAndAssert(KEY_HELLO, VALUE_HELLO_NUMERIC.toFloat())
-                myLongDictionary!!.putAndAssert(KEY_HELLO, VALUE_HELLO_NUMERIC.toLong())
-                myShortDictionary!!.putAndAssert(KEY_HELLO, VALUE_HELLO_NUMERIC.toShort())
-                myDoubleDictionary!!.putAndAssert(KEY_HELLO, VALUE_HELLO_NUMERIC.toDouble())
-                myByteDictionary!!.putAndAssert(KEY_HELLO, VALUE_HELLO_NUMERIC.toByte())
-                myDateDictionary!!.putAndAssert(KEY_HELLO, Date())
-                myObjectIdDictionary!!.putAndAssert(KEY_HELLO, ObjectId())
-                myUUIDDictionary!!.putAndAssert(KEY_HELLO, UUID.randomUUID())
-                myDecimal128Dictionary!!.putAndAssert(KEY_HELLO, Decimal128(42))
+            val objectFromRealm = transactionRealm.copyToRealm(dictionaryObject)
+            assertNotNull(objectFromRealm)
+
+            // Test primitive values
+            objectFromRealm.myStringDictionary.let { dictionary ->
+                assertNotNull(dictionary)
+                assertTrue(dictionary.containsValue(VALUE_HELLO_STRING))
+                assertTrue(dictionary.containsValue(VALUE_BYE_STRING))
+                dictionary[key] = value
+                assertTrue(dictionary.containsValue(value))
+                assertFalse(dictionary.containsValue("ANOTHER_VALUE"))
+            }
 
-                val previousByteArrayValue = myByteArrayDictionary!!.put(KEY_HELLO, VALUE_HELLO_BYTE_ARRAY)
-                assertNull(previousByteArrayValue)
-                val actualByteArrayValue = myByteArrayDictionary!![KEY_HELLO]
-                assertEquals(Byte.MIN_VALUE, actualByteArrayValue!![0])
-                assertEquals(Byte.MAX_VALUE, actualByteArrayValue[1])
+            // Test Mixed
+            objectFromRealm.myMixedDictionary.let { dictionary ->
+                assertNotNull(dictionary)
+
+                // Iterate over all Mixed types
+                for (entry: Map.Entry<String, Mixed?> in mixedEntries) {
+                    val mixedValue = entry.value
+                    if (null != mixedValue) {
+                        if (mixedValue.type != MixedType.OBJECT) {
+                            assertTrue(dictionary.containsValue(mixedValue))
+                        } else {
+                            // Cannot check if an unmanaged Mixed is in the map, it has to be managed
```

Comment:
The description isn't the best. I will address that in the refactored tests.

Mixed objects can only accept managed Realm models. When checking whether the dictionary has a Mixed instance containing an unmanaged object (which is not possible according to what I just said) we do the following:
1. get the native Mixed pointer so that we can check if the dictionary contains the value at an OS leve - if the Mixed object does not exist natively yet, we create it - which is the case here.
2. while attempting to create the native Mixed instance we use a "RealmModelOperator" (i.e. we are using a Mixed instance as a Realm model wrapper) which checks that the class of the Realm model we are providing is a RealmObjectProxy.

This fails since the Mixed instance we are using in `dictionary.containsValue(mixedValue)` contains an unmanaged Realm model.